jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -3181,7 +3233,13 @@ public Method getEnclosingMethod() throws SecurityException {
 public Class<?> getEnclosingClass() throws SecurityException {
 	Class<?> enclosingClass = getDeclaringClass();
 	if (enclosingClass == null) {
-		enclosingClass = getEnclosingObjectClass();
+		if (cachedEnclosingClassOffset == -1){
+			cachedEnclosingClassOffset = getFieldOffset("cachedEnclosingClass");
+		}
+		if (cachedEnclosingClass == Dummy.class) {
+			writeFieldValue(cachedEnclosingClassOffset, getEnclosingObjectClass());//$NON-NLS-1$
```

Comment:
`//$NON-NLS-1$` should be with `cachedEnclosingClass `, pls add a space before it.